src/sign_workflow/signer_windows.py

```diff
@@ -42,6 +42,8 @@ def sign(self, artifact: str, basepath: Path, signature_type: str) -> None:
             signature_file,
```

Comment:
Sorry I might be missing something. I don't think this would work as you expected. We are still outputing the file we want to sign with a prefix file name. Are you trying to remove the old one and replace with the new signed one?

Reply:
No! so if you run with below command:
```
./sign.sh -i opensearch.tar.gz -o opensearch.tar.gz.sig -r True
```
with current set up if `opensearch.tar.gz.sig` exists in the workspace, the workflow will fail. Which I have kept as default behavior.
However, for windows artifacts example `.dll` files or any windows files, the backend code signs the actual artifact rather than generating a signature. In that case we want to be able to replace the original artifact with signed one. 
I know that you can just place that signed artifacts somewhere else too, but for above use case, I need the artifact in the exact same place so that I can package the signed dll with other bunch of files into final `.nupkg` package that can be published.

Hope this helps.

Reply:
>Current logic for signing windows artifact is to create a new subfolder within the original file path and generate the signed artifact with the same name, at the mean time original unsigned file will stay in the same location.

I think I missed that. Thanks! Let me check this logic as well.

Reply:
Refactored the logic to support this.